File: c++/0392.cpp
// Link: https://leetcode.com/problems/is-subsequence/description/
// Runtime: 0 ms
// Memory: 6.8 MB

class Solution {
public:
  bool isSubsequence(const std::string &s, const std::string &t) {
    if (s.size() == 0)
      return true;
    else if (s.size() > t.size())
      return false;

    int i = 0, j = 0;
    while (j < static_cast<int>(t.size())) {
      if (s[i] == t[j])
        i++;
      j++;

      if (i == static_cast<int>(s.size()))
        return true;
    }

    return false;
  }
};

File: python/0392.py
# Link: https://leetcode.com/problems/is-subsequence/description/
# Runtime: 34 ms
# Memory: 16.2 MB


class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0:
            return True
        elif len(s) > len(t):
            return False

        i, j = 0, 0
        while j < len(t):
            if s[i] == t[j]:
                i += 1
            j += 1

            if i == len(s):
                return True

        return False
